tests/cpp/test_combined_inner_outer_reduction.cpp

```diff
@@ -1066,4 +1066,43 @@ TEST_F(NVFuserTest, CombinedSchedulerInnerOuterMismatch) {
   test({0});
 }
 
+// innerOuter scheduler projects buffer to inputs when there is one or more
+// outer broadcast tvs, e.g. in layer norm backward and RMS norm backward.
+// This test covers the branch where the outer broadcast tensor is not exist
+// and data type is fp32, so the buffer is not projected to inputs.
+TEST_F(NVFuserTest, CombinedSchedulerInnerOuterNoOuterBroadcastTv) {
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+
+  const int dim0 = 1024, dim1 = 2048;
+  auto tv0 = makeContigTensor(2);
+  fusion.addInput(tv0);
+  auto tv1 = sum(tv0, {1});
+  auto tv2 = broadcast(tv1, {false, true});
+  auto tv3 = add(tv2, tv0);
+  auto tv4 = sum(tv0, {0});
+  fusion.addOutput(tv3);
+  fusion.addOutput(tv4);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::randn({dim0, dim1}, options);
+  std::vector<c10::IValue> aten_inputs = {t0};
+
+  auto heuristic =
+      getInnerOuterPersistentHeuristics(fusion_ptr.get(), aten_inputs);
+  NVF_CHECK(heuristic, "InnerOuterPersistentHeuristics was not generated!");
+  NVF_CHECK(
+      !heuristic->project_persistent_buffers,
+      "Shouldn't project persistent buffers to inputs!");
+
+  FusionExecutorCache fec(std::move(fusion_ptr));
+  auto cg_outputs = fec.runFusionWithInputs(aten_inputs);
```

Comment:
Does this guarantee the inner-outer scheduler is indeed used through `FusionExecutorCache`? IIRC, the `getHeuristics` interface does not do the compile time check, does it?

Reply:
You are right, `getHeuristics` won't do compile/runtime checks. I changed to use `scheduleInnerOuterPersistentKernel + compileFusion + runFusion`, which is more straightfoward showing which scheudler is used.

we can also directly check which scheduler is used by `fec.getMostRecentKernelRuntime()                            ->schedulerHeuristics()`, either way seems fine to me.

Reply:
What did you change?

Reply:
updated, previous local change was not pushed to remote. sorry for the confusion.